### PR TITLE
Fix #5141: `inheritDoc` MUST be in camelCase.

### LINKED ISSUE #5141: phpdoc_inline_tag wrongly changes @inheritDoc (camelcase) to inheritdoc (lowercase: wrong)
## Code snippet that reproduces the problem

  ```diff
  <?php
class MyClass {
    /**
-     * {@inheritDoc}
+     * {@inheritdoc}
     */
    public function __set(string $field, $value): void
    {
    }
}
  ```

The correct tag is `inheritDoc`, in camelCase, not `inheritdoc`, in lowercase.
Source: https://docs.phpdoc.org/latest/guides/inheritance.html#the-inheritdoc-tag

This causes PHPStan not be able to recognize the `@inheritDoc` tag, causing the static analysis to fail.

### PR DESCRIPTION
**Pull request by Aerendir**
Fix #5141: `inheritDoc` MUST be in camelCase.

Source: https://docs.phpdoc.org/latest/guides/inheritance.html#the-inheritdoc-tag

### CODE REVIEW

**Comment by julienfalque** [comment]
This looks like a duplicate of #2619.

**Comment by Aerendir** [comment]
@julienfalque , I searched only the issues 😓

**Comment by Aerendir** [comment]
@julienfalque , are you interested in this PR?

**Comment by julienfalque** [comment]
I wouldn't merge it as-is: this behavior change is a BC break (from the end user point of view) so it must be opt-in, i.e. new option or new rule. I would say this is already covered by #2619 so let's close this PR as duplicate. If you're still interested in having this feature merged, please help with review on #2619 :)

**Comment by Aerendir** [comment]
@julienfalque , I don't know if it is a BC: after all this will fix a wrong docblock, so after it the code will work and inherited comments, too.
The code will continue to work and it will be simply fixed.

#2619 is way more complex than this: I sent this PR as this is something that in this moment breaks my code and my static analysis due to a wrong behavior. You PR is way more complex and I don't have the time to study it.

I'm sorry you are going to close this: this leaves me with no choices as I'm forced to decide between broke static analysis or broke code styles.

Maybe merging it now (that is way more simple and that fixes a real bug) may be a better choice? Please, think at it as this error done by PHP CS Fixer is really simple to solve in PHP CS Fixer but almost impossible to solve without fixing PHP CS Fixer. 

I'd like also to point out that this is not strictly a duplicate of #2619: #2619 has a more ample spectrum of action, whil my PR simply fixes a generic Fixer that wrongly changes the case of a PHPDoc tag.

In other words, also after the merging of @2619, the fixer I fix in this PR anyway will change the code in the wrong way and so should be anyway fixed.

In the spirit of the PR #2619 , consider this PR as a "sub-split" of #2619, to make the changes gradual (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2619#issuecomment-287769149)

**Comment by julienfalque** [comment]
I'm surprised that `{@inheritdoc}` syntax breaks static analysis, I never had such issue. Which static code analysis tools do you use?

Regarding BC: it's not about whether the fixed code continues to work (that's rather a matter of risky vs. non-risky rules). It's about a change of behavior in PHP CS Fixer that might be unexpected. The `{@inheritdoc}` syntax is very common, if PHP CS Fixer now updates to `{@inheritDoc}`, this will force users to apply the new syntax, even if they don't want to. This might also break some CI pipelines that rely on the previous behavior. This is why this should be opt-in IMO.

Also I don't think this is a bug: if I remember correctly, the initial syntax was `@inheritdoc` and was then changed to `@inheritDoc` in PSR-5 and maybe in a phpDocumentor documentation update.

Though I might reconsider splitting #2619 more, just need to find some time and motivation for that.

**Comment by Aerendir** [comment]
> I'm surprised that {@inheritdoc} syntax breaks static analysis, I never had such issue. Which static code analysis tools do you use?

I'm, too... PHPStan doesn't understand it (I use using PHPStan, Phan and Psalm together).

> Regarding BC: ... This is why this should be opt-in IMO.

I understand: you are not wrong.

> Also I don't think this is a bug

On this I don't agree too much: maybe in the beginning the syntax was in lower case, but now it is in camelCase, so it should be fixed sooner or later. IMHO.

> Though I might reconsider splitting #2619 more, just need to find some time and motivation for that.

The first is more precious than gold 😝 The second... search for it in yourself 🤣 (I'm joking: I know that once the time is passed, it is passed: I read somewhere you're not even using them anymore... so...)

In conclusion, I understand your point of view and I also agree with it: I understand it may cause issues applying the rule suddenly.

So, I'm not asking you to merge the PR.

I'm only asking you to reopen it, so it will be possible to discuss it and decide which the best approach should be and when to adhere to the official wording.

Do you think you may agree with me?

**Comment by julienfalque** [comment]
> [...] it should be fixed sooner or later. IMHO.

Agreed but I would still prefer the change to be opt-in on 2.x. Nothing prevents us from changing the default (and maybe dropping the option altogether) in 3.x though. Also, maybe other maintainers might prefer your solution, so let me reopen here to continue discussing this.

**Comment by SpacePossum** [comment]
Hi @Aerendir 

Thanks for your suggestion with this PR and putting the topic back on agenda!
I'm going to close your PR in favor of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2619 , but please let me know if this doesn't cover all your wanted to propose here and we can always reopen!
